### PR TITLE
First implementation of GemStore

### PR DESCRIPTION
**Pull request by krinart**
Hierarchy:
```
                                index.html
                               /    |      \
                              /     |       \
                             /      |        \
                            /       |         \
                           /        |          \
                          /         |           \
                         /          |            \
                        /           |             \
               product-header  product-tabs    product-gallery
                               /    |      \
                              /     |       \
                             /      |        \
                            /       |         \
                           /        |          \
                          /         |           \
                         /          |            \
                        /           |             \
       product-description   product-specs     product-reviews
                                                     |
                                                     |
                                                     |
                                                     |
                                                submit-review

TBD:

- components
- bindToController
- '<' bindings
- rework tabs to more sophisticated directives where it is only required to create pans and tabs are created automatically

### CODE REVIEW

File: app/shared/templates/submit-review.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<form name="reviewForm" ng-submit="reviewForm.$valid && submitReviewCtrl.submitReview(reviewForm, product)" novalidate>
```
Comment (krinart) [inline app/shared/templates/submit-review.html]: I'm not sure if explicitly passing form to the function is the proper way.

I'm doing it to call `$setPristine()` method in order to reset the form after new review is added.

I tried to get this form by name inside directive, but I didn't find the way to do it.

File: app/shared/directives.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        }
+    });
+
+    app.directive('productDescription', function(){
+        return {
+            restrict: 'E',
+            scope: {
+                "description": "="
+            },
+            template: "<h4>Description</h4>{{ description }}"
```
Comment (krinart) [inline app/shared/directives.js]: if I don't specify `scope` attribute here, I have an access to `product.description` where `product` belongs to the parent scope. Looks like without explicitly specifying `scope` (which creates isolated scope) I have an access to the parent scope. Is it correct?

File: app/shared/templates/submit-review.html (hunk 1 above)
Comment (kodjii) [inline app/shared/templates/submit-review.html]: It's completely ok to pass a function, nothing is wrong here.

File: app/shared/directives.js (hunk 2 above)
Comment (kodjii) [inline app/shared/directives.js]: Yes it is correct, and this is some times a correct approach, the only thing which to look over to is that changes here might affect the original, so sometimes it is safer to use <.

**Pull request merged**